include/engine/guidance/segment_list.hpp

```diff
@@ -0,0 +1,304 @@
+#pragma once
+
+#include "osrm/coordinate.hpp"
+
+#include "engine/douglas_peucker.hpp"
+#include "engine/internal_route_result.hpp"
+#include "engine/phantom_node.hpp"
+#include "engine/segment_information.hpp"
+
+#include "extractor/turn_instructions.hpp"
+
+#include <vector>
+
+// transfers the internal edge based data structures to a more useable format
+namespace osrm
+{
+namespace engine
+{
+namespace guidance
+{
+template <typename DataFacadeT> class SegmentList
+{
+  public:
+    typedef DataFacadeT DataFacade;
+    SegmentList(const InternalRouteResult &raw_route,
+                const bool extract_alternative,
+                const unsigned zoom_level,
+                const bool allow_simplification,
+                const DataFacade *facade);
+
+    const std::vector<std::uint32_t> &GetViaIndices() const;
+    std::uint32_t GetDistance() const;
+    std::uint32_t GetDuration() const;
+
+    std::vector<SegmentInformation> const &Get() const;
```

Comment:
Can we be consistent about lhs vs rhs const placement?
Three lines above it is `const T &fn() const;` here it is `T const &fn() const;`